modules/gateway/rpc.go

```diff
@@ -69,6 +69,19 @@ func (g *Gateway) RegisterRPC(name string, fn modules.RPCFunc) {
 	g.handlers[handlerName(name)] = fn
 }
 
+// UnregisterRPC unregisters an RPC and removes the corresponding RPCFunc from
+// g.handlers. Future calls to the RPC by peers will fail.
+func (g *Gateway) UnregisterRPC(name string) {
+	id := g.mu.Lock()
+	defer g.mu.Unlock(id)
+	if build.DEBUG {
+		if _, ok := g.handlers[handlerName(name)]; !ok {
+			panic("RPC is not registered: " + name)
+		}
+	}
```

Comment:
The intention of panicing here was to catch typos, but panicing is dangerous as an RPC may be registered only under certain conditions. In such a case there is no way to know through gateway's exported methods whether an RPC has been registered or not.